CONTRIBUTING.md

```diff
@@ -8,145 +8,128 @@ The following document is a rule set of guidelines for contributing.
+  - [Code structuring patterns](#code-structuring-patterns)
+  - [Functional Programming](#functional-programming)
+  - [Naming conventions](#naming-conventions)
     - [Documentation](#documentation)
-    - [Writing tests](#writing-tests)
+  - [Writing tests](#writing-tests)
 
 ## Code Contributions
 
-#### Step 1: Fork
+### Step 1: Fork
 
-Fork the project on [GitHub](git@github.com:Green-Software-Foundation/if.git)
+Fork the project on [GitHub](https://github.com/Green-Software-Foundation/if)
 
 You then have your own copy of the repository that you can change. 
 
-#### Step 2: Branch
+### Step 2: Branch
 
 Create new branch in your forked copy of the `if` repository, which will contain your new feature, fix or change. 
 
-    ```bash
-    $ git checkout -b <topic-branch-name>
-    ```
+```bash
+$ git checkout -b <topic-branch-name>
+```
 
-#### Step 3: Commit
+### Step 3: Commit
 
 Make sure git knows your name and email address:
 
-   ```bash
-   $ git config --global user.name "Example User"
-   $ git config --global user.email "user@example.com"
-   ```
+```bash
+$ git config --global user.name "Example User"
+$ git config --global user.email "user@example.com"
+```
 
-Commiting multiple files with changes on multiple resources is not allowed.
+Each commit should cover one change to one resource. You should not add multiple changes to a single commit.
 Commit message should clearly describe on which resource changes are made.
+For the commit message, we adhere to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.
+Conventional commits are organized with a type, a scope and a description. The type can be one of: 
+
+- 'feat',
+- 'fix',
+- 'docs',
+- 'chore',
+- 'style',
+- 'refactor',
+- 'ci',
+- 'test',
+- 'revert',
+- 'package',
+
+The scope is optional but should refer to the part of the codebase you are amending in the commit, e.g. `lib`, `types` etc.
+
+Here's an example of a valid commit message:
 
-Add and commit:
+```
+feat(lib): initial commit for time-sync logic
+```
 
-   ```bash
-   $ git add my/changed/files
-   $ git commit
-   ```  
-    
-Commit your changes in logical chunks. Please do not push all changes in one commit.
+or 
 
-> Run `npm run fix` before commiting for not having conflict with CI linter.
+```
+test(teads-curve): add unit test to check that error is raised on missing tdp param 
+```
 
-Please adhere to these [Commit message guidelines](#commit-message-guidelines)
-   or your code is unlikely be merged into the main project.
+Run `npm run fix` before commiting. If your commit message does not conform to the conventional commit specification or if you have not run `npm run fix` your commit will not satisfy the commitlint check.
 
-#### Step 4: Sync
+Add and commit with your commit message:
 
-Use git pull/merge to synchronize your work with the IF repository.
+```bash
+$ git add my/changed/files
+$ git commit m "<my-commit-message>"
```

Comment:
git commit m "<type-of-commit>(<my-optional-scope>): <my-commit-message>"

Reply:
```suggestion
$ git commit -m "<type-of-commit>(<my-optional-scope>): <my-commit-message>"
```